spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/ZipkinHttpSpanExtractor.java

```diff
@@ -33,7 +34,10 @@ public ZipkinHttpSpanExtractor(Pattern skipPattern) {
 	@Override
 	public Span joinTrace(SpanTextMap textMap) {
 		Map<String, String> carrier = TextMapUtil.asMap(textMap);
-		if (carrier.get(Span.TRACE_ID_NAME) == null) {
+		boolean debug = Span.SPAN_SAMPLED.equals(carrier.get(Span.SPAN_FLAGS));
+		if (debug) {
+			generateIdIfMissing(carrier, Span.TRACE_ID_NAME);
```

Comment:
TraceId is enough cause later we're generating a span if it's missing. I'll add the comment